### ISSUE TITLE
CA1508 False positive for string.IsNullOrWhitespace in pattern matching

### ISSUE THREAD

**Issue by Blackbaud-ScottFreeman** [dotnet/roslyn-analyzers#5929]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

### Describe the bug
False positive for dead code with the code shown below.

### Steps To Reproduce
```cs
WarningTest(null);
WarningTest("");
WarningTest("a");

static void WarningTest(string input)
{
    var result = input switch
    {
        null => "null",
        string s when string.IsNullOrWhiteSpace(s) => "IsNullOrWhitespace",
        _ => "default"
    };

    Console.WriteLine(result);
}
```

### Expected behavior
No dead code warning. This can be seen by the output that shows all patterns hit.

### Actual behavior
Dead code warning for string.IsNullOrWhitespace.
